### ISSUE TITLE
AdGuard blocking game (Dragonball Legends)

### ISSUE THREAD

**Issue by LoLifant** [AdguardTeam/AdGuardSDNSFilter#481]
### Steps to reproduce

1. Have active AdGuard Premium security with DNS filter.
2. Open gaming App „Dragonball Legends“.

### Expected behavior

Game should start.

### Actual behavior

After a short loading time, an error message appears even before reaching the title screen. It simply says „An error occurred“ and offers to go back to the title screen or try it again, with the former leading to (apparently) infinite loading time on a black screen and the latter just reproducing the error message. The problem started occurring a day or two ago. Everything worked before. Not sure why. The game didn‘t receive any updates from the App Store recently (only in-game content updates). Maybe whitelisting would help but since it‘s not a website I don’t know what to include into the list. When turning AdGuard off, everything works fine.

<details><summary>Screenshot: 
![25AE35C0-8B4F-442F-ACFC-151D25E4E566](https://user-images.githubusercontent.com/74019709/98290750-5c65aa00-1faa-11eb-8dfc-766dcca5dd4b.png)
</summary>

</details>

### Your environment

|   Description  |     Value    |
| -------------- | ------------ |
| DNS server in use:| AdGuard DNS
| How did you setup DNS configuration:| not sure what to write here? I just use the app and one of the dns servers already listed (see above) 
| Device model:| iPhone SE 2020
| Operating system and version:| iOS 14.1

**Comment by BlazDT** [comment]
Problematic rule according to filter log (though cannot find that rule in DNS filter):

`||35.193.`

It is blocking DNS `*.dblgnds.channel.or.jp` which results in the error.
@Alex-302 isn't that rule a bit wide?

**Comment by ameshkov** [comment]
> Problematic rule according to filter log (though cannot find that rule in DNS filter) when DNS filtering is active on my Android

Any idea where it came from? Such rules shouldn't pass validation checks for the DNS filter, and I don't see it in the filter.

**Comment by BlazDT** [comment]
Sadly no idea. It just shows up in filtering log when DNS blocking is active.

**Comment by ameshkov** [comment]
Still does?

**Comment by BlazDT** [comment]
Yep.
![Screenshot_20201106-132553](https://user-images.githubusercontent.com/34266271/98366061-a8a8fc80-2033-11eb-8e39-e03040556315.png)

**Comment by LoLifant** [comment]
So is it possible to somehow fix it by entering that: *.dblgnds.channel.or.jp into some list?

**Comment by ameshkov** [comment]
Not really.

I cannot reproduce this with AG DNS, this domain is okay.
And I am not sure why it happened to @BlazDT. @BlazDT what version of AdGuard for Android do you use?

**Comment by BlazDT** [comment]
4.0 Nightly 7

**Comment by ameshkov** [comment]
@BlazDT well, it seems you helped us find a bug with the DNS filter update in the Android version:)

**Comment by ameshkov** [comment]
@BlazDT could you please tap "Check updates" on the main screeen just in case?

**Comment by BlazDT** [comment]
I daily press on update. Though it seems DNS filter module is only getting an update even DNS is enabled as I left it enabled for a while and it is working now.

When user activates DNS filtering it should auto-update DNS filter to avoid issues like these.

**Comment by LoLifant** [comment]
Update: I‘ve experimented a bit and it seems like this problem only occurs when connected via WiFi. Everything seems to work fine on mobile data, despite having DNS enabled for both mobile and wifi.

What I also found, and what seems to be the core issue, is the tunnel setting. The problem occurs when „full tunnel without vpn icon“ is selected. The game starts up normally, both on wifi and mobile, when I change the setting to just „full tunnel“. Guess that‘s why the „full tunnel without icon“ setting mentions to be unstable in its description? Anyway, I‘ll go with that solution for now, even though I wanted the VPN icon disabled to save space on the iPhone status bar but it‘s not crucial as long as everything else works as it should.

**Comment by ameshkov** [comment]
> Guess that‘s why the „full tunnel without icon“ setting mentions to be unstable in its description?

Yeah, it sometimes causes really, really mysterious issues like this one for instance:)
